docs/reference/buildx_bake.md

```diff
@@ -361,14 +521,127 @@ EOF
 
 $ docker buildx bake --print webapp
 {
-   "target": {
-      "webapp": {
-         "context": ".",
-         "dockerfile": "Dockerfile",
-         "tags": [
-            "my-image:latest"
-         ]
+  "target": {
+    "webapp": {
+      "context": ".",
+      "dockerfile": "Dockerfile",
+      "tags": [
+        "my-image:latest"
+      ]
+    }
+  }
+}
+```
+
+Example of using variables in functions:
+
+```console
+$ cat <<'EOF' > docker-bake.hcl
+variable "REPO" {
+    default = "user/repo"
+}
+
+function "tag" {
+    params = [tag]
+    result = ["${REPO}:${tag}"]
+}
+
+target "webapp" {
+    tags = tag("v1")
+}
+EOF
```

Comment:
We were already doing this, but HEREDOC can be a bit tricky when we print it in the documentation, as it makes it hard to copy. They also don't work on Windows / PowerShell / CMD. e.g. see https://docs.docker.com/engine/reference/commandline/buildx_bake/#hcl-variables-and-functions

<img width="1022" alt="Screenshot 2021-07-28 at 11 57 02" src="https://user-images.githubusercontent.com/1804568/127303492-8cad0b08-c634-4560-9937-9b713034ed95.png">

We should consider splitting these to

1. show the file content
2. show the `docker builds bake .....` command that uses it.

Doing so also allows us to highlight the `hcl` document (if Jekyll / rouge supports the language)

```suggestion

`docker-bake.hcl`:

```hcl
variable "REPO" {
    default = "user/repo"
}

function "tag" {
    params = [tag]
    result = ["${REPO}:${tag}"]
}

target "webapp" {
    tags = tag("v1")
}
```